### PR TITLE
Make sure all tunable.step_*() are specified correctly

### PR DESCRIPTION
**Pull request by EmilHvitfeldt**
When answering https://stackoverflow.com/questions/75618633/is-this-a-bug-in-tidy-models and fixing the issue in https://github.com/tidymodels/recipes/pull/1100. I realized we are not testing this part of the code.

More specifically the output generated by `tunable.step_*()` is not tested to work, just that it has the right format. This results in 2 types of problems

- custom ranges are of invalid types https://github.com/tidymodels/recipes/pull/1100
- calls to dials functions that either doesn't exist or were renamed

This PR aims to fix that.

### CODE REVIEW

**Comment by EmilHvitfeldt** [comment]
4 of the failures are expected to be fixed by https://github.com/tidymodels/recipes/pull/1100, the last error comes from `tunable.step_impute_mean()` with:

https://github.com/tidymodels/recipes/blob/f332fcfd21b740c42b64932e3d0e9a016afb49ce/R/impute_mean.R#L236

Which as far as I can tell, `trim_amount()` was never added to {dials}

File: tests/testthat/test-dummy_multi_choice.R [hunk 1]
```diff
@@ -49,6 +49,35 @@ test_that("printing", {
...
+    step_dummy_multi_choice(all_predictors())
+  rec_param <- tunable.step_dummy_multi_choice(rec$steps[[1]])
+  expect_equal(rec_param$name, c("threshold"))
+  expect_true(all(rec_param$source == "recipe"))
+  expect_true(is.list(rec_param$call_info))
+  expect_equal(nrow(rec_param), 1)
+  expect_equal(
+    names(rec_param),
+    c("name", "call_info", "source", "component", "component_id")
+  )
```
[the remark this answers is not in the document]
Comment (hfrick) [reply tests/testthat/test-dummy_multi_choice.R:64]: applies also elsewhere :)

File: tests/testthat/test-dummy_multi_choice.R [hunk 2]
```diff
@@ -49,6 +49,35 @@ test_that("printing", {
...
+})
+
+test_that("tunable is setup to works with extract_parameter_set_dials works", {
```
Comment (hfrick) [inline tests/testthat/test-dummy_multi_choice.R:67]:
```suggestion
test_that("tunable is setup to work with extract_parameter_set_dials", {
```
and elsewhere :)

File: tests/testthat/test-dummy_multi_choice.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (hfrick) [reply tests/testthat/test-dummy_multi_choice.R:64]: and the expectation on `nrow(rec_param)` as the second expectation

File: tests/testthat/test_roll.R
Comment (hfrick) [inline tests/testthat/test_roll.R:1]: could we rename this file to `test-window.R` since it tests `step_window()` which lives in `R/window.R`?

**Review by hfrick — APPROVED** [review]
Looks good, just a few suggestions for touch-ups 👍

File: tests/testthat/test-dummy_multi_choice.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (EmilHvitfeldt) [reply tests/testthat/test-dummy_multi_choice.R:64]: This is a good point. I'm going to be doing a major testing refactoring soon, so I'm putting this on that list!

File: tests/testthat/test-dummy_multi_choice.R (hunk 2 above)
Comment (EmilHvitfeldt) [reply tests/testthat/test-dummy_multi_choice.R:67]: Done!

File: tests/testthat/test_roll.R
Comment (EmilHvitfeldt) [reply tests/testthat/test_roll.R:1]: This will done in the aforementioned testing refactor

**Pull request merged**